src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/Helpers.cs

```diff
@@ -166,7 +166,7 @@ public static RuntimeMethodInfo GetInvokeMethod(this RuntimeTypeInfo delegateTyp
         {
             Debug.Assert(delegateType.IsDelegate);
 
-            MethodInfo invokeMethod = delegateType.GetMethod("Invoke", BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+            MethodInfo invokeMethod = delegateType.GetMethod("Invoke", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly);
```

Comment:
Should the `DeclaredOnly` be there? I don't know of anything that ever derives beyond the point where `Invoke` is defined, but such a delegate could be built and would generally work, so maybe that flag should be removed.